client/client.py

```diff
@@ -0,0 +1,103 @@
+"""
+Base class which contains the basic methods to interaction with the write api of NodeBB.
+"""
+import json
+
+import requests
+import urlparse
+from django.conf import settings as django_settings
+
+
+class Client(object):
+    """
+    Client Class responsible to make connection with NodeBB and perform all calls.
+    """
+    def __init__(self):
+        self._configure()
+
+    def _configure(self):
+        self.endpoint = 'http://172.16.16.24:4568'
+        self.admin_uid = django_settings.OPENEDX_NODEBB_DISCUSSION['NODEBB_ADMIN_UID']
+        self.headers = {
+            'Authorization': 'Bearer {}'.format(django_settings.OPENEDX_NODEBB_DISCUSSION['NODEBB_API_TOKEN']),
+            'Content-Type': 'application/json'
+        }
+
+    def _request(self, method, path, **kwargs):
+        """
+        Hidden Method of Client this function will generate all of the Api Call's and return response.
+
+        Args:
+            method (str): Api call method can be Post, Put, and Delete
+            path (str): Api path to make call
+            kwargs (dictionary): All others data to post and put to Api along with call
+
+        Returns:
+            tuple: Tuple in the form (response_code, json_response) received from requests call.
+        """
+        if '_uid' not in kwargs:
+            kwargs.update({'_uid': self.admin_uid})
+
+        # `_uid` is None, there's nothing we can do at this point.
+        if kwargs['_uid'] is None:
```

Comment:
This is an unnecessary check. because you are setting the `_uid` in the kwargs above. So it will never fire.

Reply:
Done